core/arch/arm32/kernel/thread.c

```diff
@@ -73,6 +148,61 @@ thread_pm_handler_t thread_system_reset_handler_ptr;
 
 static unsigned int thread_global_lock = UNLOCK;
 
+#ifdef CFG_WITH_STACK_CANARIES
```

Comment:
Is what you're suggesting the preferred way of doing #ifdefs?
I'm happy to change it once, but I'd rather not have to change it back again because we've changed our minds.